engines/dradis-api/lib/dradis/ce/api/api_token_strategy.rb

```diff
@@ -0,0 +1,50 @@
+module Dradis::CE::API
+  class ApiTokenStrategy < ::Warden::Strategies::Base
+    attr_reader :email, :password
+
+    # This strategy should be applied when we are requesting /api/
+    def valid?
+      !!(request.path_info =~ /\A\/api\//)
+    end
+
+    def authenticate!
+      if token
+        user = User.enabled.find_by_api_token(token)
+        if user && secure_compare(user.api_token)
```

Comment:
nothing to change here, but I'm a bit stumped by the double check. I believe it's because `secure_compare` doesn't give away when the token doesn't match (as having a single DB query would)